python/tskit/tables.py

```diff
@@ -1388,22 +1392,31 @@ def _text_header_and_rows(self):
         site = self.site
         node = self.node
         parent = self.parent
+        time = self.time
         derived_state = util.unpack_strings(
             self.derived_state, self.derived_state_offset
         )
         metadata = util.unpack_bytes(self.metadata, self.metadata_offset)
-        headers = ("id", "site", "node", "derived_state", "parent", "metadata")
+        headers = ("id", "site", "node", "time", "derived_state", "parent", "metadata")
         rows = []
         for j in range(self.num_rows):
             md = base64.b64encode(metadata[j]).decode("utf8")
             rows.append(
-                "{}\t{}\t{}\t{}\t{}\t{}".format(
-                    j, site[j], node[j], derived_state[j], parent[j], md
+                "{}\t{}\t{}\t{}\t{}\t{}\t{}".format(
+                    j, site[j], node[j], time[j], derived_state[j], parent[j], md
                 ).split("\t")
             )
         return headers, rows
 
-    def add_row(self, site, node, derived_state, parent=-1, metadata=None):
+    def add_row(
+        self,
+        site,
+        node,
+        derived_state,
+        parent=-1,
+        metadata=None,
+        time=tskit.MUTATION_UNKNOWN_TIME,
```

Comment:
My only worry with this is how it'll render on RTD. In the best case, it'll render as "NaN", which will give the false impression that any NaN can be provided as the default value.

I think it's a bit safer to make the default ``None``, and just transform it before calling the ll_table version. The perf hit is trivial - anyone calling ``add_row`` in Python has said goodbye to decent perf anyway.

Reply:
Ha, good catch it rendered as `<MagicMock id='140509585067552'>` !
Fixed in https://github.com/tskit-dev/tskit/pull/672/commits/f2a1686148665737c0857c31a6190aaa5ed9e01f